Sources/Swagger.swift

```diff
@@ -116,19 +116,33 @@ struct SwaggerBuilder: Builder {
     }
 
     func build(_ swagger: SwaggerBuilder) throws -> Swagger {
+        // Pre-resolve and cache references to fix circular references:
+        SchemaBuilder.resolver.setup()
+        ParameterBuilder.resolver.setup()
+        ResponseBuilder.resolver.setup()
+        try self.definitions.values.forEach { try _ = $0.build(swagger) }
+        try self.parameters.values.forEach { try _ = $0.build(swagger) }
+        try self.responses.values.forEach { try _ = $0.build(swagger) }
+
         let paths = try Dictionary(self.paths.map { ($0, try $1.build(swagger)) })
+
         let definitions = try self.definitions.map { name, builder in
-            Structure(name: name, structure: try builder.build(swagger))
+            Structure(name: name, type: try builder.build(swagger))
         }
 
         let parameters = try self.parameters.map { name, builder in
-            Structure(name: name, structure: try builder.build(swagger))
+            Structure(name: name, type: try builder.build(swagger))
         }
 
         let responses = try self.responses.map { name, builder in
-            Structure(name: name, structure: try builder.build(swagger))
+            Structure(name: name, type: try builder.build(swagger))
         }
 
+        // Clean up resolvers:
```

Comment:
Yeah, we don't need to persist that information in memory after the resolution has finished.